src/doc/mask.cpp

```diff
@@ -129,6 +128,67 @@ void Mask::copyFrom(const Mask* sourceMask)
   }
 }
 
+void Mask::fromImage(const Image* image, const gfx::Point& maskOrigin)
+{
+  if (image) {
+    replace(image->bounds().setOrigin(maskOrigin));
+    freeze();
+    {
+      LockImageBits<BitmapTraits> maskBits(bitmap());
+      auto maskIt = maskBits.begin();
+      auto maskEnd = maskBits.end();
+
+      switch (image->pixelFormat()) {
+
+        case IMAGE_RGB: {
+          LockImageBits<RgbTraits> rgbBits(image);
+          auto rgbIt = rgbBits.begin();
+#if _DEBUG
+          auto rgbEnd = rgbBits.end();
+#endif
+          for (; maskIt != maskEnd; ++maskIt, ++rgbIt) {
+            ASSERT(rgbIt != rgbEnd);
+            color_t c = *rgbIt;
+            *maskIt = (rgba_geta(c) >= 128); // TODO configurable threshold
+          }
+          break;
+        }
+
+        case IMAGE_GRAYSCALE: {
+          LockImageBits<GrayscaleTraits> grayBits(image);
+          auto grayIt = grayBits.begin();
+#if _DEBUG
+          auto grayEnd = grayBits.end();
+#endif
+          for (; maskIt != maskEnd; ++maskIt, ++grayIt) {
+            ASSERT(grayIt != grayEnd);
+            color_t c = *grayIt;
+            *maskIt = (graya_geta(c) >= 128); // TODO configurable threshold
```

Comment:
Same case as before:
```suggestion
            *maskIt = (graya_geta(c) > 0);
```